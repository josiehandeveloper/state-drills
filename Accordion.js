import React from 'react';


class Accordion extends React.Component {
    static defaultProps = { 
        sections: [], 
    };
    state = {
        currentSectionIndex: null, 
    }
    handleSetCurrentSection = (sectionIndex) => {
        this.setState({ currentSectionIndex: sectionIndex })
    }

    renderItems(section, idx, currentSectionIndex) {
        return (
            <li className='Accordion__item' key={idx}>
                <button
                    type='button'
                    onClick={() => this.handleSetCurrentSection(idx)}
                >
                    {section.title}
                </button>
                {(currentSectionIndex === idx) && <p>{section.content}</p>}
            </li>

        )
    }


    render() {
        const { currentSectionIndex } = this.state
        const { sections } = this.props
        return (
            <ul className='Accordion'>
                {sections.map((section, idx) =>
                    this.renderItems(section, idx, currentSectionIndex)
                )}
            </ul>
        )
    }
}
            
export default Accordion;


