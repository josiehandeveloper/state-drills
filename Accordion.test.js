import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Accordion from './Accordion';
import { shallow } from 'enzyme'

describe(`Accordian Component`, () => {  
  const sectionsProp = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
     title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ];

  it('renders an empty li when section prop not supplied', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders no sections as active by default', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('component opens a clicked section', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    wrapper.find('button').at(0).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('component opens last section if multiple sections are clicked', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})