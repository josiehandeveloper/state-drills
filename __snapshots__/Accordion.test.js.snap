// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component component opens a clicked section 1`] = `
<div>
  <li
    className="Accordion-section"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <div
      className="content"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </div>
  </li>
  <li
    className="Accordion-section"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <div
      className="content"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </div>
  </li>
  <li
    className="Accordion-section"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <div
      className="content"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </div>
  </li>
</div>
`;

exports[`Accordian Component component opens last section if multiple sections are clicked 1`] = `
<div>
  <li
    className="Accordion-section"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <div
      className="content"
    >
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </div>
  </li>
  <li
    className="Accordion-section"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <div
      className="content"
    >
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </div>
  </li>
  <li
    className="Accordion-section"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <div
      className="content"
    >
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </div>
  </li>
</div>
`;

exports[`Accordian Component renders an empty li when section prop not supplied 1`] = `<div />`;

exports[`Accordian Component renders no sections as active by default 1`] = `
<div>
  <li
    className="Accordion-section"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <div
      className="content"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </div>
  </li>
  <li
    className="Accordion-section"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <div
      className="content"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </div>
  </li>
  <li
    className="Accordion-section"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <div
      className="content"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </div>
  </li>
</div>
`;
